test/perf.ts

```diff
@@ -0,0 +1,230 @@
+
+    records.push(record);
+  }
+  return records;
+}
+
+async function validate(imageName: string): Promise<string | null> {
+  try {
+    return (await execute("docker", [
+      "image",
+      "inspect",
+      imageName,
+      "--format",
+      "'{{.Id}}'",
+    ])).stdout.trim();
+  } catch {
+    // Ignore 1st failure, maybe the image does no exists locally
+  }
+
+  if (
+    imageName.startsWith("fake-") &&
+    imageName.endsWith("g") &&
+    imageName.length > 6
+  ) {
+    // attempt to create the missing fake image of ?G size
+    try {
+      await execute(join(__dirname, "..", "fake-image"), [
+        imageName.substr(5, imageName.length - 6),
+      ]);
+    } catch (err) {
+      // Ignore fake image creation failure, make a 2nd attempt to inspect
+    }
+  } else {
+    // attempt to pull the missing image
+    try {
+      await execute("docker", ["pull", imageName]);
+    } catch {
+      // Ignore pull failure, make a 2nd attempt to inspect
+    }
+  }
+
+  try {
+    return (await execute("docker", [
+      "image",
+      "inspect",
+      imageName,
+      "--format",
+      "'{{.Id}}'",
+    ])).stdout.trim();
+  } catch (err) {
+    // null image
+  }
+  // tslint:disable-next-line
+  console.error(`Error: Failed to validate ${imageName}`);
+  return null;
+}
+
+function hrdiff(hrtime: [number, number]): number {
+  const hrdiff = process.hrtime(hrtime);
+  return Math.round((hrdiff[0] * 1e9 + hrdiff[1]) / 1e6 + 1e-7) / 1e3;
+}
+
+function format(csv: boolean, record?: { [key: string]: string }): string {
+  return HEADERS.map((v, i) => {
+    return (record ? record[v] : v).toString().padEnd(csv ? 0 : padding);
+  }).join(csv ? "," : "");
+}
+
+/**
+ * command line argument parsing
+ * @returns dict of opt and pos representing option arguments as a dict
+ *  and positional as an array respectivly
+ */
+function argparse(): {
+  opt: { [key: string]: string | boolean };
+  pos: string[];
+} {
+  let index: number = 0;
+  // look for the beginning of the arguments
+  while (index < process.argv.length && process.argv[index] !== __filename) {
+    index++;
+  }
+  index++;
+  // options arguments
+  const optargs: { [key: string]: string | boolean } = {};
+  while (index < process.argv.length && process.argv[index].startsWith("-")) {
+    const av: string = process.argv[index++];
+    // end of options
+    if (av === "--") {
+      break;
+    }
+    // string value option
+    if (av.startsWith("--") && index < process.argv.length) {
+      index++;
+      optargs[av.substr(2)] = process.argv[index++];
+    }
+    // boolean value option
+    else {
+      optargs[av.substr(1)] = true;
+    }
+  }
+  // positional arguments
+  const posargs: string[] = [];
+  while (index < process.argv.length) {
+    posargs.push(process.argv[index++]);
+  }
+  return { opt: optargs, pos: posargs };
+}
+
+function main() {
+  const args = argparse();
+  if ("h" in args.opt) {
+    // tslint:disable-next-line
```

Comment:
Disable the specific rule please; `disable-next-line no-console` for example (or whatever the rule is)

Reply:
Done

Reply:
Done, again, now only for the specific lines of code